clients/include/rocblas_test.hpp

```diff
@@ -54,7 +54,19 @@
         }                                         \
     } while(0)
 
-#define EXPECT_ROCBLAS_STATUS ASSERT_EQ
+// This wraps the rocBLAS call with catch_signals_and_exceptions_as_failures().
+// By placing it at the rocBLAS call site, memory resources are less likely to
+// be leaked in the event of a caught signal.
+#define EXPECT_ROCBLAS_STATUS(STATUS, EXPECT)                               \
+    do                                                                      \
+    {                                                                       \
+        rocblas_status stat = rocblas_status(-1);                           \
+        catch_signals_and_exceptions_as_failures([&] { stat = (STATUS); }); \
+        if(stat == rocblas_status(-1))                                      \
+            return;                                                         \
+        else                                                                \
+            ASSERT_EQ(stat, (EXPECT));                                      \
+    } while(0)
```

Comment:
This change should remove most memory leaks. It puts another `catch_signals_and_exceptions_as_failures` wrapper around rocBLAS calls.